Update Fluent UI Blazor to v4.11.3

vnbaaij opened the pull request:
Update @JamesNK 

4.11.x changes FluentDataGrid to use table/tr/td instead of web components. A lot of custom CSS and JavaScript needs to react to the new elements and attributes in the HTML.

I've been through the app a number of times and tested every page that has grids. It looks ok to me, but I'm sure there are some outstanding issues. I think the best way to find them is merge and have bugs reported about dashboard issues.

---
Update @vnbaaij 
If you'd like to know more about the why behind the FluentDataGrid change, see https://github.com/microsoft/fluentui-blazor/discussions/2667

## Description

Initial work needed to move to the Fluent UI Blazor 4.11.0 release ~(not out yet)~. This is still in draft.

Fixes # (issue)

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [x] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Is this introducing a breaking change?
      - [ ] Yes
        - Link to aspire-docs issue (please use this [`breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)):
      - [ ] No
        - Link to aspire-docs issue (please use this [`doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)):
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6840)

vnbaaij committed:
Initial work for using Fluent UI Blazor v4.11.0

vnbaaij committed:
Fix grid padding inline start

vnbaaij committed:
Use preview package

vnbaaij committed:
Use Fluent UI beta.1 packages

JamesNK commented:
Stop and Play icons are no longer found:

![image](https://github.com/user-attachments/assets/29ee9926-4bd5-424a-8462-bef1e4f17dea)

They're both the filled variant. Changing variant to regular makes them resolve. Perhaps there is a bug finding any filled icons.

JamesNK committed:
Fix build

JamesNK commented:
Data grid is getting a margin-bottom from `table` rule in reboot.css:

![image](https://github.com/user-attachments/assets/452558cf-c649-43fa-8f21-27dacc09a2c1)

JamesNK commented:
First column on structured logs and traces isn't positioned correctly:

![image](https://github.com/user-attachments/assets/799678f7-fe04-4189-b4ee-cc2cd2e107da)

JamesNK commented:
Many issues on trace details:

![image](https://github.com/user-attachments/assets/8c9a716c-c908-4c29-9113-3bc428da73ee)

JamesNK commented:
Bad colors in details view:

![image](https://github.com/user-attachments/assets/748952d6-bc81-420a-ab9f-8a1608e82e95)

JamesNK commented:
As mentioned offline, there are many CSS rules that include `fluent-data-grid-*`: https://github.com/search?q=repo%3Adotnet%2Faspire%20fluent-data-grid&type=code

They're all broken and will cause bugs.

vnbaaij commented:
> Stop and Play icons are no longer found:
> 

> They're both the filled variant. Changing variant to regular makes them resolve. Perhaps there is a bug finding any filled icons.
Yes, it is indeed a bug in selecting the right assembly to look in. Will be fixed for next build

vnbaaij commented:
> Data grid is getting a margin-bottom from `table` rule in reboot.css:
> 
Setting `margin-bottom: 0px` for fluent-data-grid class (rendered table automatically always ets that class name)

vnbaaij committed:
Fixes issues found in #6840

vnbaaij committed:
Merge branch 'vnbaaij/fluentui-411' of https://github.com/dotnet/aspire into vnbaaij/fluentui-411

vnbaaij committed:
Merge main

vnbaaij committed:
Fix test

vnbaaij commented:
I've addressed most of the issues but would be good to go through a test together to see what needs to be fixed still. We do need to wait for a new version of the v4.11 packages to do so because we had to make some changes on our side as well.

vnbaaij committed:
Merge branch 'main' into vnbaaij/fluentui-411

JamesNK committed:
Fix build

JamesNK committed:
Merge

JamesNK committed:
Fix

JamesNK committed:
Update

JamesNK commented:
@vnbaaij I've updated to 4.11.1 and fixed a lot more issues. I found one regression with empty rows (see https://github.com/microsoft/fluentui-blazor/issues/3160)

JamesNK committed:
More fixes

JamesNK committed:
Fixes

JamesNK committed:
Merge branch 'main' into vnbaaij/fluentui-411

JamesNK committed:
Update

JamesNK committed:
More fixes

JamesNK committed:
Fix tests

JamesNK committed:
Source mapping

JamesNK committed:
More fixes

eerhardt commented on src/Aspire.Dashboard/Utils/DashboardUIHelpers.cs:
```diff
@@ -19,4 +23,20 @@ internal static class DashboardUIHelpers
 
     public static readonly TimeSpan ToastTimeout = TimeSpan.FromMilliseconds(5000);
+
+    public static (ColumnResizeLabels reiszeLabels, ColumnSortLabels sortLabels) CreateGridLabels(IStringLocalizer<ControlsStrings> loc)
```
```suggestion
    public static (ColumnResizeLabels resizeLabels, ColumnSortLabels sortLabels) CreateGridLabels(IStringLocalizer<ControlsStrings> loc)
```

vnbaaij committed:
Update src/Aspire.Dashboard/Utils/DashboardUIHelpers.cs

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>
